docker-compose.yml

```diff
@@ -2,16 +2,45 @@ version: '3.8'
 services:
   mysql:
     image: mysql:8.0
-    command: --default-authentication-plugin=mysql_native_password
+    restart: always
+    command:
+      --default-authentication-plugin=mysql_native_password
+      --bind-address=0.0.0.0
     cap_add:
       - SYS_NICE
     ports:
-      - '3306:3306'
+      - 3306:3306
     environment:
       - MYSQL_ROOT_PASSWORD=default_password
       - MYSQL_DATABASE=checkpoint
     volumes:
       - mysql:/var/lib/mysql
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    healthcheck:
+      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost", "-uroot", "-ppass"]
+      interval: 5s
+      timeout: 5s
+      retries: 20
+    network_mode: bridge
```

Comment:
probably it (and [here](https://github.com/checkpoint-labs/checkpoint-template/pull/4/files#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3R42)) could be removed, because Docker Compose will by default create a single network for your app, and all containers will be connected to it. This allows them to communicate with each other using the service names as hostnames.